### ISSUE TITLE
Custom Pull-to-refresh

### ISSUE THREAD

**Issue by Nico04** [marcelgarus/cached_listview#6]
With your current implementation, it seems that if I have a ListView inside a Column for instance (with a footer), the pull-to-refresh doesn't work.

```
             Column(
                children: <Widget>[
                  Expanded(
                    child: ListView(
                      children: List.generate(20, (index) => Text('$index')),
                      itemExtent: 100,
                    ),
                  ),
                  Text('Footer'),
                ],
              );
```

And if I set `hasScrollBody: false`, it throws.

I guess I would have to use the PullToRefersh directly above the ListView, inside the Expanded.
Maybe the pull-to-refresh should be optionnal or external, so we can use our custom behaviour ?
For instance I could want to implement a custom PullToRefresh component like [this](https://raw.githubusercontent.com/xuelongqy/flutter_easyrefresh/master/v2/art/image/nested_scroll_view.gif).

Thanks.

**Comment by MarcelGarus** [comment]
Having multiple nested scrolling views is kind of hard to do currently. It is easily possible though by going one abstraction layer deeper – that is, using `Sliver`s.
Currently, you'd have to instantiate a `Sliver` inside a `CachedRawBuilder`. I'm looking forward though to release a `CachedRawSliverBuilder` of some sort that can directly be used inside other slivers.

**Comment by Nico04** [comment]
I've tried a lot of combinaison of sliver inside a CustomScrollView, but I didn't succeed to enable your PullToRefresh with a footer. You have a lead to do that ?

**Comment by MarcelGarus** [comment]
Not really… The package uses the `PullToRefresh` widget from the [<kbd>pull_to_refresh</kbd>](https://pub.dev/packages/pull_to_refresh) package and I'm not an expert in how exactly it works…
Maybe you could do some further research in that direction or open a support issue over there?

**Comment by Nico04** [comment]
As I needed to change the pull-to-refresh component, I had to remove your package, so I'll will not investigate more, sorry.
Your package has a lot of potential ;)
